src/Workspaces/Core/Portable/Editing/ImportAdderService.cs

```diff
@@ -51,52 +48,15 @@ public enum Strategy
             var nodes = root.DescendantNodesAndSelf().Where(
                 n => spansTree.HasIntervalThatOverlapsWith(n.FullSpan.Start, n.FullSpan.Length));
 
-            var (importDirectivesToAdd, namespaceSymbols, context, newRoot) = strategy switch
-            {
-                Strategy.AddImportsFromSymbolAnnotations => GetImportDirectivesFromAnnotatedNodes(model, nodes, root, addImportsService, generator, cancellationToken),
-                Strategy.AddImportsFromSyntaxes => GetImportDirectivesFromSyntaxes(model, nodes, root, addImportsService, generator, cancellationToken),
-                _ => throw new InvalidEnumArgumentException(nameof(strategy), (int)strategy, typeof(Strategy)),
-            };
-
-            // If the previous call provided a new root, use it.
-            root = newRoot ?? root;
-
-            if (importDirectivesToAdd.Length == 0)
-            {
-                return document.WithSyntaxRoot(root); //keep any added simplifier annotations
-            }
+            var placeSystemNamespaceFirst = options.GetOption(GenerationOptions.PlaceSystemNamespaceFirst, document.Project.Language);
 
             if (strategy == Strategy.AddImportsFromSymbolAnnotations)
-            {
```

Comment:
the crux of the PR is the line above.  if we're only executing the 'make safe' code in when doing AddImportsFromSymbolAnnotations, then trying to contort everythign to bottleneck here is just pointless.